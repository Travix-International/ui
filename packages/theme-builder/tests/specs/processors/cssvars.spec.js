const cssvarsProcessor = require('../../../src/processors/cssvars.js');

describe('CSS Variables processor', () => {
  let doc;

  beforeEach(() => {
    doc = {
      do: {
        some: {
          string: 'aaa',
          num: '12px',
          color: '#fff'
        }
      }
    };
  });

  it('should compile doc to CSS Variables format', () => {
    const result = cssvarsProcessor.compile(doc);
    expect(result).toMatchSnapshot();
  });

  it('should compile doc to CSS Variables format with prefix', () => {
    const result = cssvarsProcessor.compile(doc, 'pre');
    expect(result).toMatchSnapshot();
  });
});
