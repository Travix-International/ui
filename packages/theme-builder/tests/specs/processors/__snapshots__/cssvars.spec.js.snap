// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS Variables processor should compile doc to CSS Variables format 1`] = `
":root {
	--do-some-string: aaa;
	--do-some-num: 12px;
	--do-some-color: #fff;
}"
`;

exports[`CSS Variables processor should compile doc to CSS Variables format with prefix 1`] = `
":root {
	--pre-do-some-string: aaa;
	--pre-do-some-num: 12px;
	--pre-do-some-color: #fff;
}"
`;
